builder/image/slugbuilder/builder/install-buildpacks

```diff
@@ -8,10 +8,18 @@ download_buildpack() {
     buildpack_name=$(basename $buildpack_url)
     buildpack_commit="$2"
 
-    git clone $buildpack_url $BUILDPACK_INSTALL_PATH/$buildpack_name
-    pushd $BUILDPACK_INSTALL_PATH/$buildpack_name 2>&1 >/dev/null
-        git checkout --quiet $buildpack_commit
-    popd 2>&1 >/dev/null
+    set +e
+    git clone --branch $buildpack_commit --depth 1 $buildpack_url $BUILDPACK_INSTALL_PATH/$buildpack_name
+    SHALLOW_CLONED=$?
+    set -e
+    if [ $SHALLOW_CLONED -ne 0 ]; then
+        rm -rf $BUILDPACK_INSTALL_PATH/$buildpack_name
```

Comment:
Since this stands out a little bit (I can see myself in 1 year asking "what's this `rm -rf` for?), would you mind adding a comment like "if the shallow clone failed partway through, clean up and try again"?

Thanks :heart: